Added docker-compose.yml

latetedemelon opened the pull request:
Should now work better "out of the box".  Testing still required.

latetedemelon committed:
Create docker-compose.yml

latetedemelon committed:
Create .env

latetedemelon committed:
Update main.py

latetedemelon committed:
Create dashboards.yml

latetedemelon committed:
Update docker-compose.yml

latetedemelon committed:
Update README.md

latetedemelon committed:
Update docker-compose.yml

latetedemelon committed:
Update .env

latetedemelon committed:
Update .env

latetedemelon committed:
Update main.py

latetedemelon committed:
Update main.py

latetedemelon committed:
Update main.py

latetedemelon committed:
Update docker-compose.yml

latetedemelon committed:
Update docker-compose.yml

latetedemelon committed:
Update main.py

latetedemelon committed:
Create entrypoint.sh

latetedemelon committed:
Update Dockerfile

latetedemelon committed:
Update docker-compose.yml

latetedemelon committed:
Update Dockerfile

latetedemelon committed:
Update Dockerfile

latetedemelon committed:
Update Dockerfile

latetedemelon committed:
Update Dockerfile

latetedemelon committed:
Update Dockerfile

latetedemelon committed:
Update Dockerfile

latetedemelon committed:
Update Dockerfile

latetedemelon committed:
Create test.txt

latetedemelon committed:
Rename influx.py to app/influx.py

latetedemelon committed:
Rename main.py to app/main.py

latetedemelon committed:
Rename myair.py to app/myair.py

latetedemelon committed:
Delete entrypoint.sh

latetedemelon committed:
Rename myair_client/__init__.py to app/myair_client/__init__.py

latetedemelon committed:
Rename myair_client/client.py to app/myair_client/client.py

latetedemelon committed:
Rename myair_client/legacy.py to app/myair_client/legacy.py

latetedemelon committed:
Rename myair_client/rest.py to app/myair_client/rest.py

latetedemelon committed:
Update docker-compose.yml

latetedemelon committed:
Rename app/influx.py to influx.py

latetedemelon committed:
Rename app/main.py to main.py

latetedemelon committed:
Rename app/myair.py to myair.py

latetedemelon committed:
Rename app/myair_client/__init__.py to myair_client/__init__.py

latetedemelon committed:
Rename app/myair_client/client.py to myair_client/client.py

latetedemelon committed:
Rename app/myair_client/legacy.py to myair_client/legacy.py

latetedemelon committed:
Rename app/myair_client/rest.py to myair_client/rest.py

latetedemelon committed:
Delete app/test.txt

latetedemelon committed:
Update docker-compose.yml

latetedemelon committed:
Update Dockerfile

vdbg commented:
Sorry for the late review; just came back from vacation where I had no Internet access. Some general comments:
* Adding another method (env based + docker compose) is a great idea, but we shouldn't remove the ability to have influx and grafana in separate docker images. I have 1 influx with a dozen docker containers writing to it, and 1 grafana with 20+ dashboards unrelated to resmed. So this should be an "in addition to" vs. an "instead of"
* We shouldn't break backwards compatibility for the config. There are existing users of this package that use the config file. Logic should be: read config file if exists, then read environment variables if exist (patching the dictionary accordingly), then fail if sections missing. This way we can have the best of both worlds.
* The program shouldn't make (direct) filesystem modifications. Writes should be limited to logs and sending stuff to influx.

vdbg commented on README.md:
```diff
@@ -11,36 +11,16 @@ Allows for importing [MyAir](https://myair.resmed.com/) data to [InfluxDB](https
-  * `sudo docker container rm myAir`
-  * ``sudo docker run -d --name myAir -v "`pwd`/config.toml:/app/config.toml" --restart=always --memory=100m resmed-influx-image``
+2. `cd resmed-influx`
+3. edit the .env with your myAir Credientals
```
typo credentials. Same line 23

vdbg commented on main.py:
```diff
@@ -11,25 +12,63 @@
+        }
+    }
+    with open(Path(__file__).with_name(CONFIG_FILE), "w") as config_file:
+        tomllib.dump(config_data, config_file)
```
Would this end up deleting an existing config? Not good if someone didn't take a backup of their config and updates to new version of container.

vdbg commented on main.py:
```diff
@@ -11,25 +12,63 @@
+
+def get_config(retry=False):
+    config_path = Path(__file__).with_name(CONFIG_FILE)
+    if config_path.is_dir():
```
don't understand when this would happen?

vdbg commented on .env:
```diff
@@ -0,0 +1,14 @@
+RESMED_LOGIN=resmed_user_email
```
we've "lost" all the documentation of the parameters. Maybe add a comment that points to the template config file?

vdbg commented on Dockerfile:
```diff
@@ -30,5 +30,6 @@ RUN  pip install --no-cache-dir -r ./requirements.txt
+RUN touch                 /app/config.toml
 
-ENTRYPOINT python main.py
+ENTRYPOINT ["python", "main.py"]
```
is this related to docker compose changes?

vdbg commented on README.md:
```diff
@@ -11,36 +11,16 @@ Allows for importing [MyAir](https://myair.resmed.com/) data to [InfluxDB](https
 
 ## Setup
 
-Choose one of these 3 methods.
```
let's keep the ability to have separate influx and grafana separately (meaning: add a new method instead of replacing existing methods)

vdbg commented:
[This](https://github.com/vdbg/resmed-influx/pull/7) PR adds support for environment variables, and removes the requirement for a config file, without breaking backwards compatibility. 

Feel free to update your PR to add docker compose functionality with updated instructions

vdbg commented:
@latetedemelon : the changes to add env var support are in

latetedemelon committed:
Update myair.py

latetedemelon committed:
Update main.py

latetedemelon committed:
Update influx.py

latetedemelon committed:
Update rest.py

latetedemelon committed:
Update legacy.py

latetedemelon committed:
Update client.py

latetedemelon committed:
Create config.py

latetedemelon committed:
Update config.py

claude committed:
Add VictoriaMetrics backend (provider=resmed)

ResMed was the only source still requiring InfluxDB. Add a VictoriaMetrics
backend so it can feed the shared platform store, selectable via
[main].backend (default 'victoriametrics', 'influx' still supported).

- victoria.py: VictoriaConnector mirroring InfluxConnector's interface;
  tracks last-imported time in a JSON state file (VM has no Flux to query
  it back), skips string fields, tags series provider=resmed
- main.py: choose backend at runtime; influxdb-client now imported lazily
- template.config.toml: new [victoria] section + backend selector
- requirements.txt: add requests; mark influxdb-client optional
- docker-compose.yml: VictoriaMetrics-based standalone stack

claude committed:
docs: link to unified platform; note VictoriaMetrics/provider change

claude committed:
Rebrand to health-influx-grafana
